compiler/src/yul/runtime/functions/data.rs

```diff
@@ -50,6 +50,18 @@ pub fn ccopym() -> yul::Statement {
     }
 }
 
+/// Load a static string from data into a newly allocated segment of memory.
+pub fn load_data_string() -> yul::Statement {
+    function_definition! {
+        function load_data_string(code_ptr, size) -> mptr {
```

Comment:
@g-r-a-n-t I know we talked about making this smarter so that one could use `load_data_string("foo")` but the problem is that `dataoffset` and `datasize` can only be used with a string literal directly. If I wanted to pass in `"foo"` as a variable I could not do `dataoffset(passed_var)` in here.

Reply:
interesting...